presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -565,15 +604,22 @@ public void setJmxExport(boolean export)
         jmxExportListener.accept(this, export);
     }
 
-    public InternalResourceGroup getOrCreateSubGroup(String name)
+    public InternalResourceGroup getOrCreateSubGroup(String name, OptionalInt dynamicSegmentPosition)
```

Comment:
This code is a little bit hard to follow. Could you please explain why is it needed to pass the `OptionalInt dynamicSegmentPosition` as a parameter? And why simple `boolean` is not enough? It feels like `createGroupIfNecessary` could be setting this boolean.